changelog.d/pa-2477.changed

```diff
@@ -0,0 +1,15 @@
+taint-mode: Historically, the matching of taint sinks has been somewhat imprecise.
+For example, `sink(ok if tainted else ok)` was flagged. Recently, we made sink-
+matching more precise for sinks like `sink(...)` declaring that any argument of
+a given function is a sink. Now we make it more precise when specific arguments of
+a function are sinks, like:
+
+```yaml
+pattern-sinks:
+- patterns:
+  - pattern: sink($X, ...)
+  - focus-metavariable: $X
+```
+
+So `sink(ok1 if tainted else ok2)`, `sink(not_a_propagator(tainted))`, and
```

Comment:
Tbh I don't understand it. It's too subtle for me.

Reply:
That's fair, if you are not used to writing taint rules it can be subtle. I'll try to make it more clear in the docs the next time I give them a pass.